Add tuterm tutorial/demo

veracioux opened the pull request:
Closes #646.

@chrisallenlane  Here's a first draft.

You can run a non-interactive demo as:
```
tuterm cheat.tut -m demo
```

You can also run an interactive tutorial as:
```
tuterm cheat.tut
```

I have nothing to add to the non-interactive demo in terms of commands. What's
left to do:

- Some polish to the demo, possibly
- Customize prompts, delays and colors based on your suggestions
- Create a script to record and upload to asciinema
- Create an animation (GIF or SVG) and put it into the README

I recorded the session and uploaded it to asciinema, so you can preview it:

[![asciicast](https://asciinema.org/a/442955.svg)](https://asciinema.org/a/442955)

Please let me know if you have any {qu,sugg}estions. Make sure to read the
TODO/NOTE comments I added to the file.

chrisallenlane commented:
Hi, @veracioux 

Sorry for the delay. I've been unexpectedly busy for the past two weeks.

This looks great! Thanks so much for the effort you've put into this. I think it's a wonderful and concise introduction to `cheat`.

I don't really have any suggestions beyond "ship it."

I see that you have this PR marked as a draft. Let me know when you're happy with it, and I'll merge it. I'm quite happy with it as-is :+1: 

Thanks!

veracioux commented:
@chrisallenlane I'm glad you're happy with it!

I've made some more changes:
- Slightly increased delays for some commands which were a bit too fast IMO
- Added an `asciinema_upload_and_create_svg.sh` which does exactly what its name says
- Added an SVG animation in the main README. I hope you don't mind that I plugged my project under the animation. I can remove it if you think it doesn't fit in there.

You should upload the demo to asciinema yourself and give me that URL so I can put it into the README, so that clicking on the svg animation takes you to asciinema. After that change, I say this is ready to be merged.
